### PR TITLE
fix: resolve null pointer exceptions when getting auth model id

### PR DESCRIPTION
**Pull request by rhamzeh**
## References

- fixes: openfga/go-sdk#16

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -710,6 +710,13 @@ func (request SdkClientReadAuthorizationModelRequest) Options(options ClientRead
 	return request
 }
 
+func (request SdkClientReadAuthorizationModelRequest) getAuthorizationModelIdOverride() *string {
+	if request.options == nil || request.options.AuthorizationModelId == nil {
```
Comment (adriantam) [inline client/client.go]: do we need to check request.options.AuthorizationModelId == nil?

File: client/client.go [hunk 2]
```diff
@@ -984,6 +995,13 @@ func (request SdkClientWriteRequest) Options(options ClientWriteOptions) SdkClie
 	return request
 }
 
+func (request SdkClientWriteRequest) getAuthorizationModelIdOverride() *string {
+	if request.options == nil || request.options.AuthorizationModelId == nil {
```
Comment (adriantam) [inline client/client.go]: similar to above, do we need to check request.options.AuthorizationModelId

File: client/client_test.go [hunk 3]
```diff
@@ -1285,6 +1376,11 @@ func TestOpenFgaClient(t *testing.T) {
 		if len(*got.Assertions) != len(*expectedResponse.Assertions) || (*got.Assertions)[0].Expectation != (*expectedResponse.Assertions)[0].Expectation {
 			t.Fatalf("OpenFgaClient.%v() = %v, want %v", test.Name, string(responseJson), test.JsonResponse)
 		}
+		// ReadAssertions without options should work
+		_, err = fgaClient.ReadAssertions(context.Background()).Execute()
+		if err != nil && err.Error() != "Required parameter AuthorizationModelId was not provided" {
```
Comment (adriantam) [inline client/client_test.go]: do we ever expect `Required parameter AuthorizationModelId was not provided`?

File: client/client_test.go [hunk 4]
```diff
@@ -1327,5 +1423,10 @@ func TestOpenFgaClient(t *testing.T) {
 		if err != nil {
 			t.Fatalf("%v", err)
 		}
+		// WriteAssertions without options should work
+		_, err = fgaClient.WriteAssertions(context.Background()).Body(requestBody).Execute()
+		if err != nil && err.Error() != "Required parameter AuthorizationModelId was not provided" {
```
Comment (adriantam) [inline client/client_test.go]: Same as above, do we ever expect 'Required parameter AuthorizationModelId was not provided'?

File: client/client_test.go (hunk 3 above)
Comment (rhamzeh) [reply client/client_test.go]: Yep, in this case because the model id was not provided

Comment (adriantam) [reply client/client_test.go]: Should we fail if err is nil then?

**Review by adriantam — APPROVED** [review]

**Pull request merged**